### PR TITLE
Mac Support: Round 1

### PR DESCRIPTION
**Pull request by RPangrle**
#373 #452 #290 

This is the first round of Mac support. It doesn't represent the finished product, but gets us at a starting point where everything is building and we can have people working on top of this without having to fork for long periods of time and try to merge some potentially large architectural changes over.

Most iOS specific code is protected by ifdefs. The larger ifdefs should get moved off to platform specific files and utilities, although some of the 1-5 line ifdefs will probably be left in place.

Some iOS only features are here and compiling on Mac, although with access to those code paths blocked. ex. Broker and Workplace Join support. In both cases a check to use those features will return NO. In both cases we need to define what those features should even look like on Mac OS X, and some of the crypto in pkeyauth will need mac specific code as the crypto calls we use on iOS aren't available on OS X. Keeping this code compiling on Mac at least ensures that we don't accidentally go backwards.

### CODE REVIEW

**Comment by RPangrle** [comment]
@kpanwar

**Comment by jasoncoolmax** [comment]
I am on it :)

File: ADAL/src/public/ADAL.h [hunk 1]
```diff
@@ -26,6 +26,21 @@ FOUNDATION_EXPORT double ADALFrameworkVersionNumber;
 //! Project version string for ADALiOSFramework.
 FOUNDATION_EXPORT const unsigned char ADALFrameworkVersionString[];
 
+#if TARGET_OS_IPHONE
+//iOS:
```
Comment (jasoncoolmax) [inline ADAL/src/public/ADAL.h]: Any reason why we use IPHONE to represent iOS? :P

Comment (RPangrle) [inline ADAL/src/public/ADAL.h]: We don't, Apple does. That's a system macro. It dates back to when iOS was called "iPhone OS"

Comment (RPangrle) [inline ADAL/src/public/ADAL.h]: Fun fact: TARGET_OS_MAC is defined as true on both iOS and Mac OS, which is why !TARGET_OS_IPHONE is used to determine if we're on a Mac or not.

Comment (jasoncoolmax) [inline ADAL/src/public/ADAL.h]: Got it, haha :)

**Comment by jasoncoolmax** [comment]
I signed off :)

File: ADAL/src/public/ADAuthenticationSettings.h [hunk 2]
```diff
@@ -41,6 +43,7 @@
 /*! The dispatch queue to be used for the asynchronous calls. */
 @property (retain) dispatch_queue_t dispatchQueue;
```
Comment (kpanwar) [inline ADAL/src/public/ADAuthenticationSettings.h]: Lets just remove it.

File: ADAL/src/ui/mac/ADCredentialCollectionController.m [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+@synthesize usernameLabel = _usernameLabel;
+@synthesize passwordLabel = _passwordLabel;
+
+-(id) init
+{
+    self = [super init];
+    if(self)
+    {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```
Comment (kpanwar) [inline ADAL/src/ui/mac/ADCredentialCollectionController.m]: ignore warnings?!?! Can use initWithNibNamed?

**Comment by kpanwar** [comment]
Sign-off with comments,

File: ADAL/src/ui/mac/ADCredentialCollectionController.m (hunk 3 above)
Comment (RPangrle) [inline ADAL/src/ui/mac/ADCredentialCollectionController.m]: #474

File: ADAL/src/public/ADAuthenticationSettings.h (hunk 2 above)
Comment (RPangrle) [inline ADAL/src/public/ADAuthenticationSettings.h]: #475

**Pull request merged**